front/src/applications/editor/tools/routeEdition/components/RouteEditionPanel.tsx

```diff
@@ -0,0 +1,314 @@
+import { ExtendedEditorContextType } from '../../editorContextTypes';
+import {
+  getCompatibleRoutesPayload,
+  getRouteEditionState,
+  getRouteGeometries,
+  routeHasExtremities,
+} from '../utils';
+import { Endpoints } from './EndPoints';
+import { SearchRoute } from './SearchRoute';
+import { RouteMetadata } from './RouteMetadata';
+
+export const RouteEditionPanel: FC<{ state: RouteEditionState }> = ({ state }) => {
+  const { t } = useTranslation();
+  const infraID = useInfraID();
+  const dispatch = useDispatch();
+  const [postPathfinding] = osrdEditoastApi.endpoints.postInfraByInfraIdPathfinding.useMutation();
+  const { setState, isFormSubmited, setIsFormSubmited } = useContext(
+    EditorContext
+  ) as ExtendedEditorContextType<RouteEditionState>;
+
+  const isNew = useMemo(
+    () => !state.entity || state.entity.properties.id === NEW_ENTITY_ID,
+    [state.entity]
+  );
+
+  const disableTransit = useMemo(() => {
+    let shouldbeDisabled = true;
+    if (state.entity.properties.release_detectors.length > 0) {
+      shouldbeDisabled = false;
+    }
+    // We can swith from empty to filled when there is a selected route in the search
+    // because we've got the data
+    else if (
+      state.optionsState.type === 'options' &&
+      !isNil(state.optionsState.focusedOptionIndex)
+    ) {
+      shouldbeDisabled = false;
+    }
+    return shouldbeDisabled;
+  }, [state]);
+
+  /**
+   * Function to save an entity.
+   * if previous is not provided, then it's a create, otherwise it's an update
+   */
+  const saveEntity = useCallback(
+    async (entity: RouteEntity, previous?: RouteEntity) => {
+      const payload = previous
+        ? {
+            update: [
+              {
+                source: previous,
+                target: entity,
+              },
+            ],
+          }
+        : {
+            create: [entity],
+          };
+
+      const result = (await dispatch<unknown>(save(infraID, payload))) as Array<{
+        railjson: RouteEntity['properties'];
+      }>;
+      setState((prev) => ({
+        ...getRouteEditionState({
+          ...entity,
+          properties: result && result[0] ? result[0].railjson : entity.properties,
+        } as RouteEntity),
+        // we keep the extremetiesEntity state, it has not changed
+        extremitiesEntity: prev.extremitiesEntity,
+      }));
+    },
+    [infraID, dispatch, setState]
+  );
+
+  /**
+   * What we do when the rigid transit checkbox changed
+   */
+  const onReleaseDetectorChange = useCallback(
+    (withDetector: boolean) => {
+      setState((prev) => {
+        let detectors: string[] = [];
+        if (
+          withDetector &&
+          prev.optionsState.type === 'options' &&
+          !isNil(prev.optionsState.focusedOptionIndex)
+        ) {
+          const selected = prev.optionsState.options[prev.optionsState.focusedOptionIndex];
+          detectors = selected.data.detectors;
```

Comment:
Doesn't change much but might help for maintainability and harmonization with the other functions

```suggestion
          const selectedCandidate = prev.optionsState.options[prev.optionsState.focusedOptionIndex];
          detectors = selectedCandidate.data.detectors;
```